primitives/state-machine/src/ext.rs

```diff
@@ -745,7 +727,7 @@ mod tests {
 		let mut ext = TestExt::new(&mut overlay, &mut offchain_overlay, &mut cache, &backend, state, None);
 		assert_eq!(
 			ext.storage_changes_root(&H256::default().encode()).unwrap(),
-			Some(hex!("bb0c2ef6e1d36d5490f9766cfcc7dfe2a6ca804504c3bb206053890d6dd02376").to_vec()),
+			Some(hex!("dfb314e4d5ddfaed1b6ef8d98d9898971bffb303a5d50e0b9c9a7bfd42ccdb90").to_vec()),
```

Comment:
This is a bit worrisome because I expected those changes to be pretty cosmetical.

Why was this changed?

Reply:
It only changed because with the public functions the exact overlay layout cannot be reproduced. It changed slightly because we cannot set the modifying extrinsic of the `EXTRINSIC_INDEX` entry itself (nor it does make sense to do so). I verified that the root is the same when I directly set the fields. So it is only caused by the changes to the tests.

Reply:
Ok, sounds plausible.